evaluation/model_evaluation.py

```
@@ -0,0 +1,1144 @@
+
+            if test_func is not None:
+                if isinstance(test_func, list):
+                    for func in test_func:
+                        func_name = func.__name__
+                        try:
+                            if asyncio.iscoroutinefunction(func):
+                                success, score, reason = await func(
+                                    model_answer, correct_answer, user_input=prompt
+                                )
+                                if not success:
+                                    raise AssertionError(reason)
```

Ruff (0.14.2)

Abstract `raise` to an inner function

(TRY301)